Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.cpp

```diff
@@ -0,0 +1,543 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzCore/Jobs/JobCompletion.h>
+#include <AzCore/Jobs/JobFunction.h>
+#include <AzCore/RTTI/TypeInfo.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Debug/EventTrace.h>
+
+#include <Atom/RHI/Factory.h>
+#include <Atom/RHI/RHIUtils.h>
+#include <Atom/RHI/ImagePool.h>
+#include <Atom/RHI/CpuProfiler.h>
+#include <Atom/RHI/RHISystemInterface.h>
+
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Pass/PassSystemInterface.h>
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RPI.Public/Shader/Shader.h>
+#include <Atom/RPI.Public/Shader/ShaderResourceGroup.h>
+#include <Atom/RPI.Public/Image/ImageSystemInterface.h>
+#include <Atom/RPI.Public/Image/AttachmentImagePool.h>
+
+#include <Atom/RPI.Reflect/Buffer/BufferAssetView.h>
+#include <Atom/RPI.Reflect/Asset/AssetUtils.h>
+
+// Hair specific
+#include <Rendering/HairGlobalSettings.h>
+#include <Rendering/HairFeatureProcessor.h>
+#include <Rendering/HairBuffersSemantics.h>
+#include <Rendering/HairRenderObject.h>
+
+#include <Passes/HairSkinningComputePass.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        namespace Hair
+        {
+            const char* HairFeatureProcessor::s_featureProcessorName = "HairFeatureProcessor";
+            uint32_t HairFeatureProcessor::s_instanceCount = 0;
+
+            HairFeatureProcessor::HairFeatureProcessor()
+            {
+                GlobalShapeConstraintsPass = Name{ "HairGlobalShapeConstraintsComputePassTemplate" };
+                CalculateStrandDataPass = Name{ "HairCalculateStrandLevelDataComputePassTemplate" };
+                VelocityShockPropagationPass = Name{ "HairVelocityShockPropagationComputePassTemplate" };
+                LocalShapeConstraintsPass = Name{ "HairLocalShapeConstraintsComputePassTemplate" };
+                LengthConstriantsWindAndCollisionPass = Name{ "HairLengthConstraintsWindAndCollisionComputePassTemplate" };
+                UpdateFollowHairPass = Name{ "HairUpdateFollowHairComputePassTemplate" };
+
+                ++s_instanceCount;
+
+                // Create right away for everyone to use
+                if (!CreatePerPassResources())
```

Comment:
This part is going to change in order to create buffers only if the passes exist and request them (hence be demand from the active pipeline).